File: projects/order-creation-queries/core/src/main/java/com/robinfood/core/entities/menu/MenuHallProductResponseEntity.java
package com.robinfood.core.entities.menu;

import lombok.Data;

import java.math.BigDecimal;
import java.util.List;

@Data
public class MenuHallProductResponseEntity {

    private final Long articleId;
    private final Long brandId;
    private final String description;
    private final BigDecimal discount;
    private final Long displayType;
    private final List<MenuProductGroupEntity> groups;
    private final Long id;
    private final String image;
    private final String name;
    private final Long parentId;
    private final Integer position;
    private final BigDecimal price;
    private final String productFlow;
    private final Long sizeId;
    private final String sku;
    private final List<String> tags;
    private final Long type;
    private final String typeName;
}

File: projects/posv2/core/src/main/kotlin/com/robinfood/core/entities/menu/MenuHallProductResponseEntity.kt
package com.robinfood.core.entities.menu

import java.math.BigDecimal

data class MenuHallProductResponseEntity(
        val articleId: Long?,
        val brandId: Long?,
        val description: String?,
        val discount: BigDecimal?,
        val displayType: Long?,
        val groups: List<MenuProductGroupEntity>?,
        val id: Long?,
        val image: String?,
        val name: String?,
        val parentId: Long?,
        val position: Int?,
        val price: BigDecimal?,
        val productCategoryId: Long?,
        val productFlow: String?,
        val sizeId: Long?,
        val sku: String?,
        val tags: List<String>?,
        val type: Long?,
        val typeName: String?
)
